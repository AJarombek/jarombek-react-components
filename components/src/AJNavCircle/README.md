### Overview

Reusable component representing a navigational arrow.

**Use cases:**

```javascript
import { AJNavCircle } from 'jarombek-react-components';

const CustomComponent = () => 
  <AJNavCircle
    direction="left"
    height="50px"
    active={false}
    onClick={f=>f} />
```

### Files

| Filename                   | Description                                                                     |
|----------------------------|---------------------------------------------------------------------------------|
| `AJNavCircle.js`           | The reusable `AJNavCircle` component.                                           |
| `index.js`                 | Index file which exports the `AJNavCircle` component.                           |
| `styles.js`                | JSS styles for the `AJNavCircle` component.                                     |
