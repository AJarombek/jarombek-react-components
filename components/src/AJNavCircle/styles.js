/**
 * Stylesheet for the AJNavCircle component using JSS.
 * Author: Andrew Jarombek
 * Date: 12/28/2019
 */

export default {
  ajNavCircle: {
    height: ({ height }) => `calc(${height} * 1.2)`,
    width: ({ height }) => `calc(${height} * 1.2)`,
    borderRadius: '50%',
    '& > figure': {
      display: 'flex',
      height: '100%',
      width: '100%',
      margin: 0,
      alignItems: 'center',
      justifyContent: 'center',
    },
  },
  active: {
    backgroundColor: '#ccc',
    '&:hover': {
      backgroundColor: '#d9d9d9',
    },
    cursor: 'pointer',
  },
  inactive: {
    backgroundColor: '#eee',
    '& image': {
      filter: 'opacity(0.5)',
    },
  },
  img: {
    height: ({ height }) => height,
    width: ({ height }) => height,
  },
  up: {
    transform: 'rotate(180deg)',
    margin: '0 0 5% 0',
  },
  down: {
    margin: '5% 0 0 0',
  },
  left: {
    transform: 'rotate(90deg)',
    margin: '0 5% 0 0',
  },
  right: {
    transform: 'rotate(270deg)',
    margin: '0 0 0 5%',
  },
};
