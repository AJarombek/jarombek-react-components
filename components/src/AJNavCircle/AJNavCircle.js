/**
 * Reusable navigation circle component which can be used on its own or part of a parent component.
 * @author Andrew Jarombek
 * @since 12/26/2019
 */

import React from 'react';
import PropTypes from 'prop-types';
import {createUseStyles} from 'react-jss';
import styles from './styles';

const useStyles = createUseStyles(styles);

const AJNavCircle = ({direction, height="15px"}) => {
  const classes = useStyles({height});

  let pictureClass = '';
  /*switch (direction) {
    case 'up':

  }*/

  return (
    <div className={classes.ajNavCircle}>
      <figure>
        <img className={pictureClass} src="../assets/arrow.png"/>
      </figure>
    </div>
  );
};

AJNavCircle.propTypes = {
  direction: PropTypes.oneOf(['up', 'down', 'left', 'right']).isRequired
};

export default AJNavCircle;
