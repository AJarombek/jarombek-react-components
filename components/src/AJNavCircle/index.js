import AJNavCircle from './AJNavCircle';

export default AJNavCircle;
