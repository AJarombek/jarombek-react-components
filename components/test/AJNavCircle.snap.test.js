import React from 'react';
import AJNavCircle from '../src/AJNavCircle/AJNavCircle';
import renderer from 'react-test-renderer';

/**
 * Snapshot test for the {@link ../../components/AJNavCircle} component.
 * @author Andrew Jarombek
 * @since 12/29/2019
 */

it('renders correctly', () => {
  const tree = renderer
    .create(<AJNavCircle height="40px" active={true} direction="right" onClick={(f) => f} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
