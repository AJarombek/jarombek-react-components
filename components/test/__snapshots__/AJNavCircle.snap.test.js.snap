// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="aj-nav-circle ajNavCircle-0-2-1 ajNavCircle-0-0-2-9 active-0-2-2"
  onClick={[Function]}
>
  <figure>
    <img
      className="nav-right right-0-2-8 img-0-2-4 img-1-0-2-10"
      src="test-file-stub"
    />
  </figure>
</div>
`;
